cocos2d/core/layers/CCLayerCanvasRenderCmd.js

```diff
@@ -91,19 +114,23 @@
 
             var bakeContext = locBakeSprite.getCacheContext();
             var ctx = bakeContext.getContext();
-            locBakeSprite.resetCanvasSize(boundingBox.width, boundingBox.height);
 
-            bakeContext.setOffset(0 - boundingBox.x, ctx.canvas.height - boundingBox.height + boundingBox.y );
             locBakeSprite.setPosition(boundingBox.x, boundingBox.y);
 
-            //visit for canvas
-            node.sortAllChildren();
-            cc.renderer._turnToCacheMode(this.__instanceId);
-            for (var i = 0, len = children.length; i < len; i++) {
-                children[i].visit(this);
+            if(this._updateCache > 0){
+                locBakeSprite.resetCanvasSize(boundingBox.width, boundingBox.height);
+                bakeContext.setOffset(0 - boundingBox.x, ctx.canvas.height - boundingBox.height + boundingBox.y );
+                //visit for canvas
+                node.sortAllChildren();
+                cc.renderer._turnToCacheMode(this.__instanceId);
+                for (var i = 0, len = children.length; i < len; i++) {
+                    children[i].visit(this);
+                }
+                cc.renderer._renderingToCacheCanvas(bakeContext, this.__instanceId);
+                locBakeSprite.transform();                   //because bake sprite's position was changed at rendering.
+                this._updateCache--;
```

Comment:
Because the first frame is empty, it is only executed by visit.
Need to enter the second time to draw a node